include/diy/mpi/collectives.hpp

```diff
@@ -32,33 +35,42 @@ namespace mpi
       MPI_Bcast(Datatype::address(x[0]),
                 x.size(),
                 Datatype::datatype(), root, comm);
+#endif
     }
 
     static request ibroadcast(const communicator& comm, T& x, int root)
     {
+#ifndef DIY_NO_MPI
       request r;
       MPI_Ibcast(Datatype::address(x),
                  Datatype::count(x),
                  Datatype::datatype(), root, comm, &r.r);
       return r;
+#endif
```

Comment:
You need to return something here. Perhaps, only `MPI_Ibcast` needs to be wrapped in the guards?

Reply:
Although, the entire request system in the no-mpi case is a bit shaky. Maybe it's best to add `DIY_UNSUPPORTED_MPI_CALL` to trigger an error. Then we can deal with it, with a specific use case in mind.

Reply:
doh, yup.